### PR TITLE
Data table sizing

### PR DESCRIPTION
**Pull request by tdraier**
## Description

This allows to use sizing properties in column definition and ensure the proper sizing are used. Adds `{ meta: grow }` table column property to use a flex grow.

Screenshot 2024-09-12 at 11 38 26

## Risk

layout issues on tables

## Deploy Plan

deploy sparkle

### CODE REVIEW

File: sparkle/src/components/DataTable.tsx [hunk 1]
```diff
@@ -118,8 +130,14 @@ export function DataTable<TData extends TBaseData>({
   }, [filter, filterColumn]);
 
   return (
-    <div className="s-flex s-flex-col s-gap-2">
-      <DataTable.Root className={className}>
+    <div
+      className={classNames(
+        "s-flex s-flex-col s-gap-2",
+        className || "",
+        widthClassName || ""
```
Comment (JulesBelveze) [inline sparkle/src/components/DataTable.tsx:137]:
```suggestion
        widthClassName
```

File: sparkle/src/components/DataTable.tsx [hunk 2]
```diff
@@ -185,7 +206,7 @@ export function DataTable<TData extends TBaseData>({
                   return null;
                 }
                 return (
-                  <DataTable.Cell key={cell.id}>
+                  <DataTable.Cell column={cell.column} key={cell.id}>
```
Comment (JulesBelveze) [inline sparkle/src/components/DataTable.tsx:209]: OOC: what's this used for?

File: sparkle/src/components/DataTable.tsx [hunk 3]
```diff
@@ -247,13 +264,23 @@ DataTable.Header = function Header({
 
 interface HeadProps extends React.ThHTMLAttributes<HTMLTableCellElement> {
   children?: ReactNode;
+  column: Column<any>; // eslint-disable-line @typescript-eslint/no-explicit-any
+  widthClassName?: string;
```
Comment (JulesBelveze) [inline sparkle/src/components/DataTable.tsx:268]: `nit`: as we have a default value we can remove the `?`

File: sparkle/src/components/DataTable.tsx [hunk 4]
```diff
@@ -247,13 +264,23 @@ DataTable.Header = function Header({
...
+  widthClassName,
+  ...props
+}: HeadProps) {
   return (
     <th
+      style={getSize(column.columnDef)}
       className={classNames(
-        "s-w-full s-py-1 s-pr-3 s-text-left s-font-medium s-text-element-800",
+        "s-py-1 s-pr-3 s-text-left s-font-medium s-text-element-800",
+        widthClassName || "",
```
Comment (JulesBelveze) [inline sparkle/src/components/DataTable.tsx:283]:
```suggestion
        widthClassName,
```

File: sparkle/src/components/DataTable.tsx [hunk 5]
```diff
@@ -118,8 +130,14 @@ export function DataTable<TData extends TBaseData>({
...
+      className={classNames(
+        "s-flex s-flex-col s-gap-2",
+        className || "",
+        widthClassName || ""
+      )}
+    >
+      <DataTable.Root>
         <DataTable.Header>
           {table.getHeaderGroups().map((headerGroup) => (
             <DataTable.Row key={headerGroup.id}>
```
Comment (JulesBelveze) [inline sparkle/src/components/DataTable.tsx:143]: We never pass `widthClassName` to `DataTable.Row`, so we could probably remove it from the `RowProps`?

**Review by JulesBelveze — APPROVED** [review]
LGTM, left some comments but feel free to merge once tackled 🦾

File: sparkle/src/components/DataTable.tsx (hunk 5 above)
Comment (tdraier) [reply sparkle/src/components/DataTable.tsx:143]: yes we do line 195

File: sparkle/src/components/DataTable.tsx (hunk 2 above)
Comment (tdraier) [reply sparkle/src/components/DataTable.tsx:208]: it's used to get the size (meta) properties from columnDef

File: sparkle/src/components/DataTable.tsx (hunk 5 above)
Comment (JulesBelveze) [reply sparkle/src/components/DataTable.tsx]: my bad, overlooked

**Pull request merged**